paimon-core/src/test/java/org/apache/paimon/table/FileStoreTableTestBase.java

```diff
@@ -1240,6 +1248,50 @@ public void testSchemaPathOption() throws Exception {
         assertThat(schemaPath).isEqualTo(tablePath);
     }
 
+    @Test
+    public void testBranchWriteAndRead() throws Exception {
+        FileStoreTable table = createFileStoreTable();
+
+        generateBranch(table);
+
+        // Write data to branch1
+        try (StreamTableWrite write = table.newWrite(commitUser);
+                StreamTableCommit commit = table.newCommit(commitUser, BRANCH_NAME)) {
+            write.write(rowData(2, 20, 200L));
+            commit.commit(1, write.prepareCommit(false, 2));
+        }
+
+        // Validate data in branch1
```

Comment:
We also need to verify that the data of the `main` is not affected by `branch1`

Reply:
I've already added it, thanks for the suggestion. Please open ci.